### PR TITLE
Set publication name from code if none given

### PR DESCRIPTION
**Pull request by kenoir**
Fixes a problem with usages being refused if no publication name is given.

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala [hunk 1]
```diff
@@ -41,13 +41,20 @@ case class PrintUsageMetadata(
   type LongElement = List[(String,Long)]
   type IntElement = List[(String,Int)]
 
+  val calculatedPublicationName =
+    if (publicationName.trim.isEmpty) publicationCode match {
```
Comment (akash1810) [inline common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala]: `trim.isEmpty` does that mean some publication codes are whitespace? :fearful:

**Comment by NickPapacostas** [comment]
*sheepish* :+1:

File: common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala [hunk 2]
```diff
@@ -41,13 +41,20 @@ case class PrintUsageMetadata(
...
+      case "gdn" => "Guardian"
+      case "obs" => "Observer"
+      case _ => "Unknown Publication"
+    } else publicationName
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala]: Minor: should this also be trimmed? Or should we trim prior to storing the data?

File: common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala (hunk 1 above)
Comment (kenoir) [inline common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala]: @akash1810 I think they might sometimes be whitespace at the moment, this is within our control though, @blishen is looking `t it.

**Comment by kenoir** [comment]
I'm going to see if we've resolved this on the other side before merging.

File: common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala (hunk 2 above)
Comment (kenoir) [inline common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala]: This is protecting specifically against the case where we try and store only whitespace in a dynamo field which borks, so I don't think it matters if we trim as long as the fields not only whitespace.

Ideally we don't do this at all.

Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/model/usage/PrintUsageMetadata.scala]: :+1: (dynamo non-support for empty strings still boggles my mind)

**Comment by kenoir** [comment]
Looks like we fixed this on the other side, so closing.

**Pull request closed without merging**